src/main/scala/xiangshan/backend/decode/DecodeUnit.scala

```diff
@@ -982,6 +982,21 @@ class DecodeUnit(implicit p: Parameters) extends XSModule with DecodeUnitConstan
     decodedInst.exceptionVec(illegalInstr) := io.fromCSR.illegalInst.vsIsOff
   }
 
+  // decode for SoftPrefetch instructions (prefetch.w / prefetch.r / prefetch.i)
+  val isSoftPrefetch = inst.OPCODE === BitPat("b0010011") && inst.FUNCT3 === BitPat("b110") && inst.RD === 0.U
+  val isPreW = isSoftPrefetch && inst.RS2 === 3.U(5.W)
+  val isPreR = isSoftPrefetch && inst.RS2 === 1.U(5.W)
+  val isPreI = isSoftPrefetch && inst.RS2 === 0.U(5.W)
+
+  when(isPreW || isPreR || isPreI){
+    decodedInst.selImm := SelImm.IMM_S
+    decodedInst.fuType := FuType.ldu.U
+    decodedInst.canRobCompress := false.B
+    decodedInst.fuOpType := Mux(isPreW,
+                                LSUOpType.prefetch_w,
+                                Mux(isPreR, LSUOpType.prefetch_r, LSUOpType.prefetch_i))
+  }
```

Comment:
+ Merge this when block into other when-elsewhen-otherwise block would be more readable. But it's ok for now.
+ Use `Mux1H` instead of `Mux`es will get better PPA, when the select conditions have no priorities. 
It's ok to merge this PR. Two issues above can be done in future PRs